### DISCUSSION TITLE
Setting up OpenVPN in new Env

### DISCUSSION THREAD

**Discussion by charan-digitaltrust** [gruntwork-io/knowledge-base#321] [Help] [answered]
Hi all,
I have created a brand new Env and was able to bring up all the AWS resources using the Terragrunt scripts that we have except for the OpenVPN. We are no longer subscribers of Gruntworks and so we rely on modules hosted locally in our bitbucket. 

When I am executing the `terragrunt apply` on the OpenVPN `terrgrunt.hcl` file, I am seeing the error below. 
```
Error: Expected to find a Root Device name for AMI (ami-062b821853cfec0ee), but got none

  on .terraform/modules/openvpn/modules/openvpn-server/main.tf line 40, in resource "aws_launch_configuration" "openvpn":
  40: resource "aws_launch_configuration" "openvpn" {
```
  
 From a little bit of exploring, it looks like it depends on the packer (`openvpn-server.json`) file. However, it requires the github token in order to pull the gruntworks cli tools. I looked up for the similar CLI tools in our bitbucket library containing all the Gruntworks related repos, but could find nothing. 
  
Could you show me a way in which I can bring up OpenVPN server in my new Env?

**Comment by yorinasub17** [comment, marked as answer]
The AMIs are built using the deploy runner, and are managed by the pipeline through changes to the corresponding build script. Specifically, in the `infrastructure-live` repo you should see a folder with bash scripts under the path `shared/REGION/_regional/amis`. Here is [an example](https://github.com/gruntwork-io/terraform-aws-service-catalog/tree/master/examples/for-production/infrastructure-live/shared/us-west-2/_regional/amis).

Note that for AMI building when expanding to a new region, you have two options:

- Create a new AMI in the target region by copying and updating the ami build script to target the new region.
- Follow the instructions in [this knowledge base entry](https://github.com/gruntwork-io/knowledge-base/discussions/171) to copy the AMI across all the regions you need.
